contentreview-impl/impl/src/java/org/sakaiproject/turnitin/util/TurnitinLTIUtil.java

```diff
@@ -0,0 +1,182 @@
+		turnitinSite = serverConfigurationService.getString("turnitin.lti.site", "!turnitin");
+	}
+	
+	public boolean makeLTIcall(int type, String urlParam, Map<String, String> ltiProps){
+		try {
+	        
+			HttpClientParams httpParams = new HttpClientParams();
+			httpParams.setConnectionManagerTimeout(60000);
+			HttpClient client = new HttpClient();
+			client.setParams(httpParams);
+			client.getParams().setParameter("http.protocol.content-charset", "UTF-8");
+			
+			//Map<String,String> extra = new HashMap<String,String> ();
+			String extra = "";
+			
+			String defUrl = formUrl(type, urlParam);
+			if(defUrl == null){
+				log.error("Error while getting TII LTI url.");//TODO params
+				return false;
+			}
+			
+			PostMethod method = new PostMethod(defUrl);
+			//ltiProps = BasicLTIUtil.signProperties(ltiProps, defUrl, "POST", said, secret, null, null, null, null, null, extra);
+			ltiProps = BasicLTIUtil.signProperties(ltiProps, defUrl, "POST", said, secret, null, null, null, null, extra);
+			if(ltiProps == null){
+				log.error("Error while signing TII LTI properties.");//TODO params
+				return false;
+			}
+			
+			for (Entry<String, String> entry : ltiProps.entrySet()) {
+				String key = entry.getKey();
+				String value = entry.getValue();
+				if (value == null)
+					continue;
+				method.addParameter(key,value);
+			}
+			
+			int statusCode = client.executeMethod(method);//TODO process values
+			if(statusCode == 400){
+				log.warn("Status 400: Bad request: " + defUrl);
+				log.warn("LTI props " + ltiProps.toString());
+				return false;
+			} else if(statusCode == 200){//OJO para el submit el estado correcto es 200
+				log.debug("Status 200: OK request: " + defUrl);
+				log.debug("LTI props " + ltiProps.toString());
+				log.debug(method.getResponseBodyAsString());
+				return true;
+			} else if(statusCode == 302){
+				log.debug("Successful call: " + defUrl);
+				log.debug("LTI props " + ltiProps.toString());
+				log.debug(method.getResponseBodyAsString());
+				return true;
+			} else {
+				log.warn("Not controlled status: " + statusCode + " - " + method.getStatusText());
+				log.debug("LTI props " + ltiProps.toString());
+				log.debug(method.getResponseBodyAsString());
+			}
+		
+		} catch (Exception e) {
+			log.error("Exception while making TII LTI call " + e.getMessage());//TODO addparams
+			return false;
+	    }
+		
+		return true;//TODO
+	}
+	
+	public String getGlobalTurnitinLTIToolId(){
+		if(globalId == null){
+			log.debug("Setting global TII LTI tool id");
+			List<Map<String, Object>> tools = ltiService.getToolsDao("lti_tools.site_id = '"+turnitinSite+"'", null, 0, 0, turnitinSite);
+			if ( tools == null || tools.size() < 1  || tools.size() > 1) {
+				if(tools == null)
+					log.warn("No tools found");
+				else
+					log.warn("Found: " + tools.size());				
+				log.error("getGlobalTurnitinLTIToolId: wrong global TII LTI tool configuration");
```

Comment:
It's complemented with the previous warnings, it's a way to highlight the id is not set.